### PR TITLE
Feat/burn token

### PR DESCRIPTION
**Pull request by PaddyMc**
### Description

An issuer can burn emoney tokens

closes: #87

### How to test

- `make start-dev`
- `make seed`
- `cosmos-cashd query bank total --output json | jq` => check the supply
- `cosmos-cashd tx issuer burn-token 9997 --from validator --chain-id cash -y`

### CODE REVIEW

File: x/issuer/types/msg.go [hunk 1]
```diff
@@ -61,3 +61,53 @@ func (msg MsgCreateIssuer) GetSigners() []sdk.AccAddress {
...
+	if msg.Amount == 0 {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "please enter a fee greater than 0")
+	}
+
+	return nil
+
+}
+
+func (msg MsgBurnToken) GetSignBytes() []byte {
+	panic("IBC messages do not support amino")
```
Comment (noandrea) [inline x/issuer/types/msg.go:103]: Q: is it panic-ing a bit strong here?

File: x/issuer/types/msg.go [hunk 2]
```diff
@@ -61,3 +61,53 @@ func (msg MsgCreateIssuer) GetSigners() []sdk.AccAddress {
...
+}
+
+// GetSigners implements sdk.Msg
+func (msg MsgBurnToken) GetSigners() []sdk.AccAddress {
+	accAddr, err := sdk.AccAddressFromBech32(msg.Owner)
+	if err != nil {
+		panic(err)
```
Comment (noandrea) [inline x/issuer/types/msg.go:110]: Q: is it panic-ing a bit strong here?

File: x/issuer/keeper/msg_server.go [hunk 3]
```diff
@@ -74,3 +74,45 @@ func (k msgServer) CreateIssuer(
 
 	return &types.MsgCreateIssuerResponse{}, nil
 }
+
+// CreateIssuer creates a new e-money token issuer
+func (k msgServer) BurnToken(
```
Comment (noandrea) [inline x/issuer/keeper/msg_server.go:79]: please fix godoc

File: x/issuer/keeper/msg_server.go [hunk 4]
```diff
@@ -74,3 +74,45 @@ func (k msgServer) CreateIssuer(
...
+			types.ErrIssuerFound,
+			"issuer does not exists",
+		)
+	}
+
+	issuerToken := sdk.NewCoins(sdk.NewInt64Coin(issuer.Token, int64(msg.Amount)))
+
+	receipent, _ := sdk.AccAddressFromBech32(msg.Owner)
+
+	// send tokens from module to issuer
```
Comment (noandrea) [inline x/issuer/keeper/msg_server.go:97]: Q: not sure about this comment, is it correct?

File: x/issuer/types/msg.go (hunk 1 above)
Comment (alessio) [reply x/issuer/types/msg.go:103]: Mmm... I don't think so. This looks to me an unrecoverable error, thus panic() seems to fit well?

File: x/issuer/types/msg.go (hunk 2 above)
Comment (alessio) [reply x/issuer/types/msg.go:110]: Ditto

File: x/issuer/client/cli/tx.go [hunk 5]
```diff
@@ -66,9 +67,42 @@ func NewCreateIssuerCmd() *cobra.Command {
...
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			accAddr := clientCtx.GetFromAddress()
+			accAddrBech32 := accAddr.String()
+
+			fee, _ := strconv.ParseInt(args[0], 0, 32)
```
Comment (noandrea) [inline x/issuer/client/cli/tx.go:85]: please rename variable `fee` since it does not represent the actual value

**Review by noandrea — APPROVED** [review]
LGTM

**Pull request merged**